app/client/components/consent/PrivacySettings.tsx

```diff
@@ -285,16 +284,29 @@ export class PrivacySettings extends Component<{}, State> {
   }
 
   private buildState(iabVendorList: ParsedIabVendorList): void {
+    if (iabVendorList && iabVendorList.purposes) {
+      // TODO: Trigger error
+      // tslint:disable-next-line: no-console
+      console.log("ERROR: iabVendorList not present");
+    }
     // tslint:disable-next-line: no-object-mutation
     this.iabVendorList = iabVendorList;
 
-    if (iabVendorList && iabVendorList.purposes) {
-      const iabPurposes = iabVendorList.purposes.reduce((acc, purpose) => {
+    const iabStr = readIabCookie();
+
+    let iabPurposes = {};
+    if (iabStr) {
+      const iabData = new ConsentString(iabStr);
+      iabPurposes = iabVendorList.purposes.reduce((acc, purpose) => {
```

Comment:
Question: `iabVendorList` contains the latest list of vendors, but the consent string only contains the vendors at the time it was saved. I assume `iabData.isPurposeAllowed` will return false for any new vendor ID's that aren't present in the iabString? Do we want to cross check any new vendor ID's with the vendor IDs we have saved in the IAB string and if they're not present set them to `true` - as we always force the vendor to `true`?